src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -34,7 +34,9 @@ case class Instance(
   val runSpecId: PathId = instanceId.runSpecId
   val isLaunched: Boolean = state.condition.isActive
 
-  def isReserved: Boolean = state.condition == Condition.Reserved
+  // An instance has to be considered as Reserved if at least one of its tasks is Reserved.
+  def isReserved: Boolean = tasksMap.values.exists(_.status.condition == Condition.Reserved)
```

Comment:
does this change mean that instance might exist where it's condition is not reserved BUT isReserved now returns true?

Wouldn't a cleaner solution be to change the logic in a way that such instance will still have state condition set to reserved?

Reply:
Yes. Please refer to the tickets. Probably `Instance.state.condition` will be dropped soon, though I am not sure yet. It is the last big refactoring related to PVs for pods, that's to be done yet. As it is implemented now, instance/task status updates handling is wrong, if an instance has two or more tasks. We could live with the current implementation in case of ephemeral pods, but since now we introduce support for PVs pods it's become serious, and due to some kind of regression on Mesos side, which actually made this race condition happened, I am posting here just a quick fix before starting the long-time-ago-planned task handling overhaul.

Basically, in case of stateful pods, Last-Write-Wins strategy is just wrong when it comes to handling terminal status updates.

Reply:
makes sense, thanks 👍